### ISSUE TITLE
support any http options

### ISSUE THREAD

**Issue by microcai** [chriskohlhoff/urdl#1]
http options is implemented as number of classes. Is there any reason not to use a  k/v based container ?
If I want to add more options, I _HAVE_ to change the source code. That makes urdl less adaptable to regular uses.

**Comment by WoLfulus** [comment]
+1

**Comment by canselcik** [comment]
Agreed. I wish there was another `non-libcurl` HTTP client for C++. Customizing the host header requires modifying the source code.
